bins/src/bin/datadog-static-analyzer-server.rs

```diff
@@ -51,6 +51,25 @@ impl Fairing for CORS {
     }
 }
 
+#[rocket::get("/shutdown")]
+fn shutdown_get(server_configuration: &State<ServerConfiguration>) -> Status {
+    if server_configuration.is_shutdown_enabled {
+        Status::NoContent
+    } else {
+        Status::Forbidden
+    }
+}
+
+#[rocket::post("/shutdown")]
+fn shutdown_post(server_configuration: &State<ServerConfiguration>, shutdown: Shutdown) -> Status {
+    if server_configuration.is_shutdown_enabled {
+        shutdown.notify();
+        Status::Ok
+    } else {
+        Status::Forbidden
+    }
+}
+
```

Comment:
Similar, please add a comment to explain the logic as this is not documented elsewhere.